typing/typecore.ml

```diff
@@ -5692,6 +5746,11 @@ and type_expect_
                          match lid.txt with
                              Longident.Lident txt -> { txt; loc = lid.loc }
                            | _ -> assert false)
+        | Val_mut -> begin
+            match path with
+            | Path.Pident id -> Texp_mutvar {loc = lid.loc; txt = id}
+            | _ -> assert false
```

Comment:
I know I wrote this, but I think it would be better to use `Misc.fatal_error` here.  A good rule of thumb, which I didn't understand at the time, is that `assert false` is fine when the invariant is enforced locally to the relevant code, but `Misc.fatal_error` is preferrable when the invariant in question relies on other functions or files, because someone might change those other places to violate the invariant in the future and then at least we get a helpful error message. (This rule is not consistently followed in older code in the compiler, but it's good to use in new code)